src/device/device_ledger.cpp

```diff
@@ -204,14 +218,47 @@ namespace hw {
       MDEBUG( "Device "<<this->id <<" Destroyed");
     }
 
+    /* ======================================================================= */
+    /*  LOCKER                                                                 */
+    /* ======================================================================= */ 
+    
+    //automatic lock one more level on device ansuring the current thread is allowed to used it
+    #define AUTO_LOCK_CMD() \
+      /* lock both mutexes without deadlock*/ \
+      std::lock(device_locker, command_locker); \
+      /* make sure both already-locked mutexes are unlocked at the end of scope */ \
+      std::lock_guard<std::recursive_mutex> lock1(device_locker, std::adopt_lock); \
+      std::lock_guard<std::mutex> lock2(command_locker, std::adopt_lock)
```

Comment:
Please use boost locks, not std. Apparently std locks are buggy on... some ARM IIRC. hyc knows the details.

Reply:
ok